platform/security/security-token-storage/token-storage-impl/src/main/java/org/codice/ddf/security/file/token/storage/TokenInformationFactory.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.file.token.storage;
+
+import static org.codice.ddf.security.token.storage.api.TokenStorage.ACCESS_TOKEN;
+import static org.codice.ddf.security.token.storage.api.TokenStorage.DISCOVERY_URL;
+import static org.codice.ddf.security.token.storage.api.TokenStorage.REFRESH_TOKEN;
+import static org.codice.gsonsupport.GsonTypeAdapters.MAP_STRING_TO_OBJECT_TYPE;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.codice.ddf.security.token.storage.api.TokenInformation;
+import org.codice.ddf.security.token.storage.api.TokenInformationImpl;
+import org.codice.gsonsupport.GsonTypeAdapters.LongDoubleTypeAdapter;
+
+public class TokenInformationFactory {
```

Comment:
❓ I got a little confused by the name, since this isn't strictly reflecting the factory pattern. Rather than just being responsible for creating `TokenInformation` objects, this class also has `getJson` and `removeToken` methods. But if those `getJson` methods also return String versions of `TokenInformation` objects, the factory name would make more sense to me. If not, maybe a name like `TokenInformationUtil` would be better (but man, I *hate* having `util` everywhere).